### PR TITLE
Update documentation and prepare for 1.4 release

### PR DESCRIPTION
**Pull request by mattsfuller**
Update documentation on hardware and software requirements
Add release notes for 1.4
Bump version to 1.4

### CODE REVIEW

**Comment by mattsfuller** [comment]
@cawallin @rschlussel Could you please review?

File: docs/software-requirements.rst [hunk 1]
```diff
@@ -2,20 +2,6 @@
 Software Requirements
 =====================
 
-**Operating System**
```
Comment (rschlussel) [inline docs/software-requirements.rst]: I would leave the Operating System requirements in. Any flavor of linux is probably fine and possibly OS X too, but we haven't tested on anything else.  And we know there would be issues on windows.

**Comment by rschlussel** [comment]
Also could you make the commit message shorter so that it actually fits on one line.

**Comment by akshatnair** [comment]
Do we have a link somewhere in PA docs that talks about prerequisites for Presto or a link for Presto docs that specifies it?

**Comment by cawallin** [comment]
looks good % rebecca's comment

**Comment by rschlussel** [comment]
oh yes, otherwise lgtm

**Comment by cawallin** [comment]
Please also update the docs to say 1.4 and possibly add more release notes -- especially a note that this version of presto-admin works with presto <= 0.148

**Comment by mattsfuller** [comment]
@akshatnair we will add the docs in Presto that specifies the other stuff

File: docs/release/release-1.4.rst [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+===========
+
+This release works for Presto versions 0.116-0.148.
+
+* Add package uninstall support
+* Add --nodeps option to indicate if the server install/uninstall should ignore dependencies
+* Fix config files to be owned by the presto user and not accessible to other users
+* Update and add more Presto configuration defaults
+* Fix bug to use JAVA 8 for server upgrade
+* Various documentation improvements
```
Comment (cawallin) [inline docs/release/release-1.4.rst]: Doesn't need to be added

File: docs/release/release-1.4.rst [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+Release 1.4
+===========
+
+This release works for Presto versions 0.116-0.148.
+
+* Add package uninstall support
+* Add --nodeps option to indicate if the server install/uninstall should ignore dependencies
+* Fix config files to be owned by the presto user and not accessible to other users
+* Update and add more Presto configuration defaults
+* Fix bug to use JAVA 8 for server upgrade
```
Comment (cawallin) [inline docs/release/release-1.4.rst]: suggest: "Use proper Java version for server upgrade"

(that's what it was right?)

File: prestoadmin/__init__.py [hunk 4]
```diff
@@ -20,7 +20,7 @@
 
 from fabric.api import env
 
-__version__ = '1.4-SNAPSHOT'  # Make sure to update setup.py too
+__version__ = '1.4'
```
Comment (cawallin) [inline prestoadmin/__init__.py]: you need to update setup.py too :)

**Comment by cawallin** [comment]
looks good % comments

File: prestoadmin/__init__.py (hunk 4 above)
Comment (rschlussel) [inline prestoadmin/__init__.py]: and leave the comment in please

Comment (mattsfuller) [inline prestoadmin/__init__.py]: I couldn't find where this was referencing in setup.py. I assumed it was outdated..
What specifically do I need to update?

Comment (mattsfuller) [inline prestoadmin/__init__.py]: lol. nevermind. I had the wrong file open.

**Pull request closed without merging**